packages/terra-pills/src/SelectablePills.jsx

```diff
@@ -0,0 +1,337 @@
+   */
+  onSelect: PropTypes.func,
+  /**
+   * Callback function triggered on click/keypress of the roll-up pill when group of pills is collapsed.
+   */
+  onSelectRollUp: PropTypes.func,
+};
+
+const defaultProps = {
+  ariaLabelledBy: undefined,
+  ariaDescribedBy: undefined,
+  children: undefined,
+  isCollapsed: false,
+  onSelect: undefined,
+  onSelectRollUp: undefined,
+};
+
+const SelectablePills = (props) => {
+  const {
+    ariaLabel,
+    ariaLabelledBy,
+    ariaDescribedBy,
+    children,
+    intl,
+    isCollapsed,
+    onRemove,
+    onSelect,
+    onSelectRollUp,
+    ...customProps
+  } = props;
+
+  const theme = React.useContext(ThemeContext);
+  const selectablePillsRef = useRef();
+  const focusNode = useRef(0);
+  const currentPill = useRef();
+  const rollupCount = useRef(children.length);
+  const prevPillsCount = useRef(0);
+  const [containerTabindex, setContainerTabindex] = useState('-1');
+  const [updatedChildren, setUpdatedChildren] = useState(children);
+  const isRollUpRemoved = useRef(false);
+
+  const generateRollUp = useCallback(() => {
+    const startIndex = RollUpUtils.handleRollUp(selectablePillsRef);
+
+    rollupCount.current = (children.length - startIndex);
+    const reducedArray = React.Children.toArray(children).splice(0, startIndex);
+    setUpdatedChildren(reducedArray);
+  }, [children]);
+
+  const generateExpansion = useCallback(() => {
+    setUpdatedChildren(children);
+  }, [children]);
+
+  useLayoutEffect(() => {
+    if (isCollapsed) {
+      generateRollUp();
+    } else {
+      generateExpansion();
+    }
+  }, [children, generateExpansion, generateRollUp, isCollapsed]);
+
+  const setTabIndex = (val) => {
+    const currentNode = currentPill.current ? selectablePillsRef.current.querySelector(`[id=${currentPill.current}]`) : null;
+    if (currentNode) {
+      currentNode.setAttribute('tabIndex', val);
+    }
+  };
+
+  const focusCurrentNode = () => {
+    const currentNode = currentPill.current ? selectablePillsRef.current.querySelector(`[id=${currentPill.current}]`) : null;
+    if (currentNode) {
+      currentNode.focus();
+    }
+  };
+
+  useEffect(() => {
+    const pills = [...selectablePillsRef.current.querySelectorAll('[data-terra-pill]')];
+    const pillCount = pills.length;
+    const rollUpPill = selectablePillsRef.current.querySelector('[data-terra-rollup-pill]');
+    if (rollUpPill) {
+      pills.push(rollUpPill);
+    }
+
+    if (pills.length > 0) {
+      for (let i = 0; i < pills.length; i += 1) {
+        pills[i].setAttribute('tabindex', '-1');
+      }
+    }
+
+    // needs to be verified - lp052179
+    if (isCollapsed && prevPillsCount.current === pillCount) {
+      focusNode.current = 0;
+    }
+
+    // focus the first pill( be it pill or a roll up pill)
+    if (pills.length > 0 && pills.length > focusNode.current) {
+      currentPill.current = pills[focusNode.current].id;
+      setTabIndex('0');
```

Comment:
Can we declare the these values as a constant ?

const tabbingOn  = 0
const tabbingOff = -1

Reply:
`setTabIndex('0')` is actually a function.  It sets "-1" for the pill that is about to lose focus and "0" for the pill that is about to receive focus. 

[Here](https://github.com/cerner/terra-framework/pull/1442/files#diff-30d1befc1f5f254f8705585f759e7da122677fbeb6b8f3ec84ff694b83b53bd6R107) is a reference to it .

Reply:
Yup, what i was mentioning above is instead of passing the hardcoded value (0,-1 in this case) everytime. can we declare it as a constant and pass ?